modules/example/src/smithy4s/example/BucketName.scala

```diff
@@ -5,7 +5,10 @@ import smithy4s.syntax._
 
 object BucketName extends Newtype[String] {
   val id: smithy4s.ShapeId = smithy4s.ShapeId("smithy4s.example", "BucketName")
-  val underlyingSchema : smithy4s.Schema[String] = string
+  val hints : smithy4s.Hints = smithy4s.Hints(
+    id,
+  )
+  val underlyingSchema : smithy4s.Schema[String] = string.withHints(hints)
   val schema : smithy4s.Schema[BucketName] = bijection(underlyingSchema, BucketName(_), (_ : BucketName).value)
```

Comment:
Wasn't sure if this was correct or not to have the hints on `underlyingSchema`.. I am thinking they should actually be on the `schema` following the `bijection`?

Reply:
I think on the whole thing, yeah (not `underlyingSchema`)

Reply:
It shouldn't make a difference :)

Reply:
Actually,@kubukoz is correct. On the `schema`, we have a chance of writing interpreters that retain the information of the id of the underlying schema;

Reply:
Actually maybe not. I think it's fine as it is : the hints should be applied on the "underlyingSchema". That is because we do not use the newtypes for collections, except when they are used as hints themselves. However, we still want the underlying schema of collections to contain the hints.

Reply:
(We should really have some smoke tests for this kind of thing 😓 )

Reply:
I thought to have this on the big thing so that the implementation of the Schematic would carry it over to the wrapped schema in a `Hinted` or something

Reply:
> I thought to have this on the big thing so that the implementation of the Schematic would carry it over to the wrapped schema in a Hinted or something

Yeah, that is not incorrect, but having it on the small thing makes more sense, because otherwise you wouldn't have the `id` for collection shapes.

Reply:
ahh I see now